files/anitya_cron.py

```diff
@@ -34,21 +34,27 @@ def indexed_listings():
             continue
 
 
-def projects_by_feed(session):
+def projects_by_feed(session, by_ecosystem):
     """ Return the list of projects out of sync, found by feed listings.
 
     If a new entry is noticed and we don't have a project for it, add it.
     """
     for name, homepage, backend, version in indexed_listings():
-        project = anitya.lib.model.Project.get_or_create(
-            session, name, homepage, backend)
+        if by_ecosystem:
+            ecosystem = anitya.lib.model.Backend.by_name(session, backend).default_ecosystem
+            if ecosystem is None:
+                continue
+            project = anitya.lib.model.Project.get_or_create_by_name_and_ecosystem(
+                session, name, homepage, ecosystem.name, backend)
+        else:
+            project = anitya.lib.model.Project.get_or_create(
+                session, name, homepage, backend)
         if project.latest_version == version:
             LOG.debug("Project %s is already up to date." % project.name)
         else:
             yield project
 
 
-
```

Comment:
This change is good for PEP-8, but I recommend making it in a separate PEP-8 commit since it is unrelated to the purpose of this commit.